controller/login.go

```diff
@@ -61,62 +58,45 @@ func NewLoginController(service *goa.Service, auth *login.KeycloakOAuthProvider,
 
 // Authorize runs the authorize action.
 func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
-	authEndpoint, err := c.configuration.GetKeycloakEndpointAuth(ctx.Request)
-	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"err": err,
-		}, "unable to get Keycloak auth endpoint URL")
-		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get Keycloak auth endpoint URL")))
-	}
-
-	tokenEndpoint, err := c.configuration.GetKeycloakEndpointToken(ctx.Request)
+	authEndpoint, err := c.configuration.GetAuthEndpointLogin(ctx.RequestData.Request)
 	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"err": err,
-		}, "unable to get Keycloak token endpoint URL")
-		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get Keycloak token endpoint URL")))
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
-
-	entitlementEndpoint, err := c.configuration.GetKeycloakEndpointEntitlement(ctx.Request)
+	locationURL, err := RedirectLocation(ctx.Params, authEndpoint)
 	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"err": err,
-		}, "unable to get Keycloak entitlement endpoint URL")
-		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, errs.Wrap(err, "unable to get Keycloak entitlement endpoint URL")))
+		return jsonapi.JSONErrorResponse(ctx, errors.NewInternalError(ctx, err))
 	}
+	ctx.ResponseData.Header().Set("Location", locationURL)
+	return ctx.TemporaryRedirect()
+}
 
-	brokerEndpoint, err := c.configuration.GetKeycloakEndpointBroker(ctx.Request)
+func RedirectLocation(params url.Values, location string) (string, error) {
```

Comment:
Looks like this is being used in the `controller` , we needn't have it as an exported method then.

Reply:
Agree.
Done.